File: 第一期/01/二分法倒序.py
def sort(nums):
    n = len(nums)
    for i in range(1, n):
        target = nums[i]
        l, r = 0, i - 1
        while l <= r:
            mid = l + (r - l) // 2
            if target > nums[mid]:
                r = mid - 1
            else:
                l = mid + 1
        for j in range(i - 1, l - 1, -1):
            nums[j + 1] = nums[j]
        nums[l] = target
    return nums

def main():
    n = int(input())
    nums = []
    for x in range(n):
        nums.append(int(input(x)))
    sorted_nums = sort(nums)
    for x in sorted_nums:
        print(x, end=" ")

File: 第一期/01/二分法倒序.cpp
#include<iostream>
#include<vector>
using namespace std;
int main() {
    int i, n;
    cin>>n;
    cout<<endl;
    vector<int> nums(n);
    for(i = 0; i < n; i++) {
        cin>>nums[i];
    }
    for(i = 1; i < n; i++) {
        int target = nums[i];
        int l = 0, r = i - 1;
        while(l <= r) {
              int mid = l + (r - l) / 2;
              if(target > nums[mid]) r = mid - 1;
              else l = mid + 1;
        }
        for(int j = i - 1; j >= l; j--) {
            nums[j + 1] = nums[j];
        } 
        nums[l] = target;
    }
    for(int i = 0; i < n; i++) {
        cout<<nums[i]<<" ";
    }
    return 0;
}
